### PR TITLE
OCPBUGS-30162: Introduce --issuer-url flag in oc login

### PR DESCRIPTION
**Pull request by ardaguclu**
Normally, oc login extracts the issuer url from API server but in cases where multiple oidc providers exist, user may want to explicitly specify issuer url to authenticate. That's why, this PR introduces `--issuer-url` flag in oc login to be passed to `oc get-token` oidc cred exec plugin of oc.

### CODE REVIEW

**Comment by ardaguclu** [comment]
Failures are not related;
/retest

File: pkg/cli/login/login.go
Comment (stlaz) [inline pkg/cli/login/login.go:193]:
```go
oidcOptionsSet := o.OIDCClientID != "" || o.OIDCClientSecret != "" || len(o.OIDCExtraScopes) > 0 || o.OIDCIssuerURL != ""

if len(o.OIDCExecPluginType) == 0 && oidcOptionsSet {
```

File: pkg/cli/login/login.go
Comment (stlaz) [inline pkg/cli/login/login.go:100]: validate that the flag is HTTPS

File: pkg/cli/login/loginoptions.go
Comment (stlaz) [inline pkg/cli/login/loginoptions.go:341]: IMO this must always be set

**Review by stlaz — CHANGES_REQUESTED** [review]

File: pkg/cli/login/loginoptions.go
Comment (ardaguclu) [reply pkg/cli/login/loginoptions.go:341]: If this is always be set, why are we extracting from oauth-authorization-server endpoint?

File: pkg/cli/login/login.go
Comment (ardaguclu) [reply pkg/cli/login/login.go]: Changed.

File: pkg/cli/login/login.go
Comment (ardaguclu) [reply pkg/cli/login/login.go:100]: Added.

File: pkg/cli/login/loginoptions.go
Comment (ardaguclu) [reply pkg/cli/login/loginoptions.go:341]: From the oc point of view, this doesn't matter too much and if in the future, it is decided that users always have to set `issuer-url`, it is easy to add a validation. I think, we can move on as is.

Comment (stlaz) [reply pkg/cli/login/loginoptions.go:341]: > If this is always be set, why/what are we extracting from oauth-authorization-server endpoint?
ideally we shouldn't extract anything from there

**Comment by stlaz** [comment]
/hold
until we're clear on the path we're taking
/lgtm

**Comment by xingxingxia** [comment]
Failed e2e jobs are due to one MCO bug being fixed by `https://github.com/openshift/os/pull/1455`

**Comment by xingxingxia** [comment]
Failed e2e jobs were due to one MCO bug being fixed by https://github.com/openshift/os/pull/1455 .
Given `https://github.com/openshift/os/pull/1455` is emerged hours ago, let me issue `retest`:
/retest

**Comment by ardaguclu** [comment]
I don't think there is anything we can do apart from brute-force retesting;
/retest

**Comment by xingxingxia** [comment]
I have marked `https://issues.redhat.com/browse/OCPBUGS-30149` as Release Blocker to reminder related owners resolve (including verify) it faster to unblock our PR tests.

**Comment by xingxingxia** [comment]
I manage to launch cluster successfully with latest build 4.16.0-0.nightly-2024-03-05-105513 without `https://issues.redhat.com/browse/OCPBUGS-30149` error now `Cluster operator machine-config Degraded is True with RequiredPoolsFailed: Unable to apply 4.16.0-0.nightly-2024-03-04-155853: error during syncRequiredMachineConfigPools: [context deadline exceeded, failed to update clusteroperator: [client rate limiter Wait returned an error: context deadline exceeded, error MachineConfigPool master is not ready, retrying. Status: (pool degraded: true total: 3, ready 0, updated: 0, unavailable: 3)]]`. Congrats! Retest again:

/retest

**Pull request merged**
